rdkit/Chem/PandasTools.py

```diff
@@ -440,54 +440,70 @@ def ChangeMoleculeRendering(frame=None, renderer='PNG'):
       frame._repr_html_ = types.MethodType(defPandasRepr, frame)
 
 
+
+
+
+
 def LoadSDF(filename, idName='ID', molColName='ROMol', includeFingerprints=False,
             isomericSmiles=True, smilesName=None, embedProps=False, removeHs=True,
-            strictParsing=True):
-  '''Read file in SDF format and return as Pandas data frame.
-    If embedProps=True all properties also get embedded in Mol objects in the molecule column.
-    If molColName=None molecules would not be present in resulting DataFrame (only properties
-    would be read).
-    '''
-  if isinstance(filename, str):
-    if filename.lower()[-3:] == ".gz":
-      import gzip
-      f = gzip.open(filename, "rb")
+            strictParsing=True, sanitize=False):
+    '''Read file in SDF format and return as Pandas data frame.
+      If embedProps=True all properties also get embedded in Mol objects in the molecule column.
+      If molColName=None molecules would not be present in resulting DataFrame (only properties
+      would be read).
+      '''
+    if isinstance(filename, str):
+        if filename.lower()[-3:] == ".gz":
+            import gzip
+            f = gzip.open(filename, "rb")
+        else:
+            f = open(filename, 'rb')
+        close = f.close
     else:
-      f = open(filename, 'rb')
-    close = f.close
-  else:
-    f = filename
-    close = None  # don't close an open file that was passed in
-  records = []
-  indices = []
-  for i, mol in enumerate(
-      Chem.ForwardSDMolSupplier(f, sanitize=(molColName is not None), removeHs=removeHs,
-                                strictParsing=strictParsing)):
-    if mol is None:
-      continue
-    row = dict((k, mol.GetProp(k)) for k in mol.GetPropNames())
-    if molColName is not None and not embedProps:
-      for prop in mol.GetPropNames():
-        mol.ClearProp(prop)
-    if mol.HasProp('_Name'):
-      row[idName] = mol.GetProp('_Name')
-    if smilesName is not None:
-      try:
-        row[smilesName] = Chem.MolToSmiles(mol, isomericSmiles=isomericSmiles)
-      except:
-        log.warning('No valid smiles could be generated for molecule %s', i)
-        row[smilesName] = None
-    if molColName is not None and not includeFingerprints:
-      row[molColName] = mol
-    elif molColName is not None:
-      row[molColName] = _MolPlusFingerprint(mol)
-    records.append(row)
-    indices.append(i)
-
-  if close is not None:
-    close()
-  RenderImagesInAllDataFrames(images=True)
-  return pd.DataFrame(records, index=indices)
+        f = filename
+        # don't close an open file that was passed in(molColName is not None)
+        close = None
+    records = []
+    indices = []
+    for i, mol in enumerate(
+        Chem.ForwardSDMolSupplier(f, sanitize=False, removeHs=removeHs,
+                                  strictParsing=strictParsing)):
+        if not sanitize:
```

Comment:
I'm confused by this logic, this calls SanitizeMol if sanitize==True, what is the intent?

Reply:
Yes it should say ```if sanitize:``` instead ```if not sanitize```
I think it's better let the user define whether sanitize or not, when loading an SDF file.